include/deal.II/base/utilities.h

```diff
@@ -1030,27 +1030,20 @@ namespace Utilities
 {
   template <int N, typename T>
   inline T
-  fixed_power(const T n)
+  fixed_power(const T x)
   {
-    Assert(N >= 0, ExcNotImplemented());
-    switch (N)
-      {
-        case 0:
-          return dealii::internal::NumberType<T>::value(1);
-        case 1:
-          return n;
-        case 2:
-          return n * n;
-        case 3:
-          return n * n * n;
-        case 4:
-          return n * n * n * n;
-        default:
-          T result = n;
-          for (int d = 1; d < N; ++d)
-            result *= n;
-          return result;
-      }
+    static_assert(
+      !std::is_integral<T>::value || (N > 0),
+      "The non-type template parameter N must be a non-negative integer for integral type T");
```

Comment:
@bangerth I have added a `static_assert`.

Reply:
Even better!

Reply:
Well, it could be zero if an integer.

Reply:
It's getting late. Fixed.

Reply:
The testers confirm that we still need `Assert` here since we have a branch where we call the function with `-N` as template parameter (and we can't use `if constexpr`).

Reply:
@masterleinad I would really prefer to have the `static_assert` in place. What do you think of the current variant?